adobe/react-spectrum
# fix(#3842): DnD Nested Drop Regions example can't focus child drop target

Problem is with how we determine the dropTarget within the DragManager onFocus event handler. We need to prioritize the event.target matching a validDropTarget over a validDropTarget containing the event.target, otherwise the parent element may match first.


Closes #3842 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 3842](https://github.com/adobe/react-spectrum/issue/3842).

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/c42fb35488449eeae9fcb38139f02336ad9bc550/storybook/index.html?path=/story/drag-and-drop--nested-drop-regions&providerSwitcher-express=false&providerSwitcher-toastPosition=bottom
2. Move keyboard focus to the draggable element labelled "Drag me".
3. Press Enter key to start drag and drop operation.
4. Focus should move to the parent of the two nested drop targets.
5. Press Tab key to move focus to the child drop target.
6. Focus should move to the child drop target so that a user can complete the drag and drop operation using the keyboard alone.
7. Press Enter to complete the drag and drop operation.

## 🧢 Your Project:

Adobe/Accessibility
